storybookjs/storybook
# Add a `noMatchSnapshots` option.

See (https://github.com/storybooks/storyshots/pull/93). 

Issue:

The idea is that early on in a project, when things are changing a lot, snapshot tests are not particularly useful, but it's still helpful to "smoke test" all stories to ensure they run without error.

## What I did

I added a `noMatchSnapshots` option to just setup a test that runs each story, without checking snapshots.

## How to test

1. Create a project, install storyshots but pass `initStoryshots({ noMatchSnapshots: true })`.
2. Add a story, run `jest`. Should pass.
3. Change the component's output, run `jest` again. Should still pass
4. Make the component error in the render function, run `jest` again. Should fail.

## Notes:

Perhaps it would make sense to just allow passing a custom `test` function, to cover off cases like this and https://github.com/storybooks/storyshots/pull/76? Then `storyshots` is more of a generic tool to integrate storybook with Jest.

```js
initStoryshots({
  test(renderedStory) {
     // In my case, just render without looking at the result
     renderer.create(renderedStory).toJSON()

     // For #76, shallow render + snapshot
     const tree = shallowToJson(shallow(renderedStory))
     expect(tree).toMatchSnapshot()
  }
});
```
